tiatoolbox/wsicore/wsireader.py

```diff
@@ -3267,6 +3283,9 @@ class docstrings for more information.
         if interpolation in [None, "none"]:
             interpolation = None
 
+        if interpolation == "optimise" and self.mode == "bool":
```

Comment:
For features, it can be chosen by the user what to select as it depends on the scenario.
For binary masks, it should be bool otherwise it will generate unintended output.